### PR TITLE
This introduces the concept of doppelgangers 

### PR DESCRIPTION
**Pull request by bschofield-va**
that allow real users with ID.me accounts be associated with test patient data.

I could really use help testing this. I've tried all the cases that I can think of, but I'd appreciate some good independent verification that I have not broken anything or introduced a possible security vulnerability.

### CODE REVIEW

**Comment by bschofield-va** [comment]
FYI @dtournour-va @alecbert-va @aparcel-va

File: run-local.sh [hunk 1]
```diff
@@ -75,7 +75,7 @@ docker run \
   -e "KONG_LOG_LEVEL=info" \
   -e "KONG_PROXY_ACCESS_LOG=/dev/stdout" \
   -e "KONG_PROXY_ERROR_LOG=/dev/stderr"\
-  -e "KONG_PLUGINS=request-termination,response-transformer,health-apis-token-validator,health-apis-static-token-handler,health-apis-patient-registration" \
+  -e "KONG_PLUGINS=request-termination,response-transformer,health-apis-token-validator,health-apis-static-token-handler,health-apis-patient-registration,health-apis-doppelganger" \
```
Comment (jblefkove-va) [inline run-local.sh]: Are the plugins listed in any particular order?

File: kong/plugins/health-apis-doppelganger/handler.lua [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+-- support staff) in incomming HTTP requests with a known test patient user
+-- Responses will reverse the translation so that links or ICNs in the response
+-- can be presented in terms of the original request.
+--
+-- An example:
+-- I am a developer with an ID.me account but no records at the VA. My ICN is 999.
+-- This plugin is configured such that 999 is a registered doppelganger for test
+-- patient 123. Using the Apple Health app, I make a request for /Condition?patient=999
+-- This plugin will request Condition?patient=123 instead. Just prior to returning the
+-- response,  all occurrences of patient 123 are replaced with 999 to allow links
```
Comment (jblefkove-va) [inline kong/plugins/health-apis-doppelganger/handler.lua]: Is the replacement based on patient references specifically, or is just a simple string replacement of `123` with `999` in the response?

File: kong/plugins/health-apis-doppelganger/handler.lua [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+-- allow access to test data.
+-- This plugin will replace known 'doppelganger' ICNs (typically ICNs of developers or
+-- support staff) in incomming HTTP requests with a known test patient user
+-- Responses will reverse the translation so that links or ICNs in the response
+-- can be presented in terms of the original request.
+--
+-- An example:
+-- I am a developer with an ID.me account but no records at the VA. My ICN is 999.
+-- This plugin is configured such that 999 is a registered doppelganger for test
+-- patient 123. Using the Apple Health app, I make a request for /Condition?patient=999
```
Comment (jblefkove-va) [inline kong/plugins/health-apis-doppelganger/handler.lua]: Does this happen for patient reads as well?

File: kong/plugins/health-apis-doppelganger/handler.lua (hunk 2 above)
Comment (ilaflamme-va) [reply kong/plugins/health-apis-doppelganger/handler.lua]: @jblefkove-va 
> This plugin assumes that ICNs are unique enough to not naturally occur in text.
For example, the ICN 1017283132V631076 is easily recogized and replaced with
1011537977V693883 using simple string replacement techniques.

I saw this; maybe it clears things up for you?

File: kong/plugins/health-apis-doppelganger/handler.lua (hunk 3 above)
Comment (bschofield-va) [reply kong/plugins/health-apis-doppelganger/handler.lua]: yes

File: kong/plugins/health-apis-doppelganger/handler.lua (hunk 2 above)
Comment (bschofield-va) [reply kong/plugins/health-apis-doppelganger/handler.lua]: yes, simple string replacement

File: run-local.sh (hunk 1 above)
Comment (bschofield-va) [reply run-local.sh]: No particular order. Execution order is determined by the `PRIORITY` of each plugin.

**Pull request merged**